terraform/app/modules/cloudfront/versions.tf

```diff
@@ -2,7 +2,7 @@ terraform {
   required_providers {
     aws = {
       source                = "hashicorp/aws"
-      version               = ">= 3.45.0"
+      version               = ">= 5.29.0"
```

Comment:
For consistency with the other services, could you please:
- pin the version exactly `version = "5.29.0"` everywhere 
- add `-upgrade` to terraform inits